Add Loqed `last_event` and misc diagnostic sensors

t3hk0d3 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

- Adds `last_event` diagnostic sensor entity. This is very useful for automations based on source of event - for example if lock was closed from inside/outside or locking was triggered from the cloud, for example. More info on event types is available here: https://support.loqed.com/en/articles/6127856-loqed-local-bridge-api-integration
- Adds misc diagnostic sensors, such as `Battery type`, `Battery voltage`, `Wifi Strength`. This information is available through Loqed Bridge API, but not available thru Home Assistant integration. This information was requested here: https://github.com/cpolhout/loqed_custom_component/issues/8

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

t3hk0d3 committed:
Loqed: Fix typo

t3hk0d3 committed:
Loqed: Add extra diagnostic sensor entities

frenck commented on homeassistant/components/loqed/lock.py:
```diff
@@ -76,6 +76,11 @@ async def async_open(self, **kwargs: Any) -> None:
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return custom state attributes."""
+        return {EXTRA_ATTRIBUTE_LAST_EVENT: self._lock.last_event}
```
I think it would be better if this was separate sensor (with an enum), that supports translations to make it look better in the UI.

t3hk0d3 replied:
My only concerns with these approach are following:
- there might be a potential inconsistency between separate Last event entity and Lock entity states, because of async nature of entity updates
- I don't feel confident enough that a new states won't be added/changed to the LOQED firmware later.

frenck replied:
> there might be a potential inconsistency between separate Last event entity and Lock entity states, because of async nature of entity updates

As this uses a coordinator, that argument will be the same for the suggested attribute in this PR and with the separate sensor.

> I don't feel confident enough that a new states won't be added/changed to the LOQED API/firmware later.

Then we will discover that, as an invalid value will raise and we can add the translation that is needed after that.

../Frenck

t3hk0d3 replied:
Oke. Fair point. 👍

I'll move it to separate entity.

Thanks for feedback!

t3hk0d3 replied:
Done. Please check again :)

t3hk0d3 committed:
Loqed: Add `last_event` sensor entity

mikewoudenberg commented:
This looks like a nice addition to the integration. Thanks @t3hk0d3 !

mikewoudenberg reviewed (approved):
Current state looks good to me

frenck reviewed (changes requested):
@t3hk0d3 Tests are failing, can you please take a look?

Thanks! 👍 

../Frenck

t3hk0d3 commented:
> @t3hk0d3 Tests are failing, can you please take a look?
> 
> Thanks! 👍
> 
> ../Frenck

Thank you for spotting! Test failed because of last-minute changes (I've moved `battery_type` enum values to lowercase, because `hassfest` was complaning about it in `strings.json`).

Should be fixed now.

frenck commented on homeassistant/components/loqed/sensor.py:
```diff
@@ -12,16 +14,36 @@
+        key="last_event",
+        translation_key="last_event",
+        options=EVENT_TYPES,
+        transform_value=lambda data: data if data != "" else "unknown",
```
The result of an unknown item in an enum is already "unknown" automatically, this logic is thus not needed.

```suggestion
```

t3hk0d3 replied:
Unfortunately default value is `""` (empty string). It causes an exception.

frenck replied:
In case, we need to map an empty string to `None` (as in a `NoneType`), not `unknown`.

t3hk0d3 replied:
Makes sense. I'll update PR.

t3hk0d3 replied:
Should be fixed.

frenck commented on homeassistant/components/loqed/sensor.py:
```diff
@@ -12,16 +14,36 @@
-SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
-    SensorEntityDescription(
+
+@dataclass
+class LoqedSensorEntityDescription(SensorEntityDescription):
+    """Dataclass describing LOQED sensor entities."""
+
+    transform_value: Callable[[StateType], StateType] = lambda data: data
+
+
```
```suggestion
```

frenck commented on homeassistant/components/loqed/sensor.py:
```diff
@@ -30,13 +52,41 @@
+    LoqedSensorEntityDescription(
+        key="battery_type",
+        translation_key="battery_type",
+        transform_value=lambda data: str(data).lower(),
```
Instead of transforming, it might be better to define a `value_fn` that provides the actual value.

t3hk0d3 replied:
I've considered it first too, but it is needed only in two cases.

frenck replied:
Not really, it would actually improve all cases, as it would get rid of ``getattr` usage (which causes typing to be lost).

t3hk0d3 replied:
That makes sense. Will update.

t3hk0d3 replied:
Should be fixed.

frenck commented on homeassistant/components/loqed/sensor.py:
```diff
@@ -12,31 +14,83 @@
+SENSORS: Final[tuple[LoqedSensorEntityDescription, ...]] = (
+    LoqedSensorEntityDescription(
+        key="last_event",
+        value_fn=lambda data: data.last_event if data.last_event != "" else None,  # type: ignore[attr-defined]
```
```suggestion
        value_fn=lambda data: data.last_event if data.last_event else None,  # type: ignore[attr-defined]
```

frenck replied:
`data` is a dict, right?

`StatusMessage` is defined as a `TypedDict`, something is off here requiring to ignore typing because of that?

t3hk0d3 replied:
For some reason mypy triggers warning `StatusMessage has not attribute 'last_event'`.

frenck replied:
Correct, as the TypedDict wasn't extended, also, it is a dict, but you are accessing it as an object. Tests pass, as the tests seems to have a mismatch in the mock (as the mock created objects, but the library seems to return dictionaries?)

Something is fishy here :D

frenck replied:
But it is Typed as a dictionary:

<https://github.com/home-assistant/core/blob/9741380cc09592b74eb34da11f66e48f4706a05f/homeassistant/components/loqed/coordinator.py#L51>

t3hk0d3 replied:
@frenck I can only speculate what was initial idea behind this type setup. Probably `StatusMessage` was meant to be used as interface? 🤷🏻

t3hk0d3 replied:
@mikewoudenberg Would you mind to comment this issue pls?

mikewoudenberg replied:
The data coordinator is initialised with with data coming back from the bridge, which is a StatusMessage. The lock property on the coordinator is a loqed.Lock. Hope that sheds some light

frenck replied:
Not really? Is it an object or a dictionary?

mikewoudenberg replied:
The sensors read data from the `lock` property on the datacoordinator so the return type of the data property on the LoqedSensor class should be `loqed.Lock`, which is an object

t3hk0d3 replied:
@mikewoudenberg thanks. I've updated PR :)

@frenck pls check again. It should be solved now.

t3hk0d3 commented:
@frenck Issue with typing has been solved. Please check again. 

Thx.

mikewoudenberg reviewed (approved):
LGTM

t3hk0d3 commented:
@frenck Can i ask to check this PR again? I understand you must be busy, and there is also a lot of other good PRs needed attention. However this PR unlocks few critical features, and already stuck around for a while. 

Have a good holidays! And happy New Year!

t3hk0d3 commented:
![pretty please](https://static.wikia.nocookie.net/shrek/images/8/84/Cute_Eyes.jpeg/revision/latest?cb=20180904165231)

dougiteixeira committed:
Bump aioesphomeapi to 22.1.0 (#111435)

cdce8p committed:
Update caldav to 1.3.9 (#111429)

cdce8p committed:
Update guppy3 to 3.1.4.post1 (#111430)

cdce8p committed:
Update freezegun to 1.4.0 (#111268)

abmantis committed:
Fix uneeded state update in Idasen Desk sensor (#111433)

dougiteixeira committed:
Esphome text sensor device class (#111057)

PoppyPop committed:
Return group unit of measurement when device_class is None (#110973)

* Groups: Return units when device_class is None

* Fixes

* Mods

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

CFenner committed:
Add sensors for energy storage system to ViCare integration (#106600)

* add sensors for vitocharge

* add further sensors

* add unig getters

* remove icons

* Update strings.json

* handle percent uom

* handle unknown uom device class mapping

* add device class to sensor

* add sensor options

* Apply suggestions from code review

* fix

* align sensor naming

* add feed-in and consumption sensors for pcc

* Update strings.json

* battery symbol

* fix format

* remove obsolete device classes

* add icons

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* refactor

* modify lambda

* use helper

* Apply suggestions from code review

* Apply suggestions from code review

* Update strings.json

* Update sensor.py

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

astrandb committed:
Add @astrandb as myuplink codeowner (#111449)

Add codeowner

Moustachauve committed:
Add TechnoVE auto charge switch (#109093)

* Add binary sensors to TechnoVE integration

* Add unit tests for TechnoVE binary sensors

* Implement PR feedback for TechnoVE

* Limit to appropriate sensors in TechnoVE tests

* Removed leftover code

* Implement feedback in TechnoVE PR #108938

* Add auto-charge switch to TechnoVE

* Improve TechnoVE test_switches to be consistent with other platforms

* Regenerate test_switch.ambr snapshot

* Add binary sensors to TechnoVE integration

* Add unit tests for TechnoVE binary sensors

* Implement PR feedback for TechnoVE

* Limit to appropriate sensors in TechnoVE tests

* Implement feedback in TechnoVE PR #108938

* Add auto-charge switch to TechnoVE

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fix conflict merge issue

* Implement feedback from TechnoVE PR #109093

* Use TechnoVESwitchDescription

* Remove None from is_on in TechnoVE switches

* Update homeassistant/components/technove/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Remove unneeded code.

* Update test_switch.ambr

* Update TechnoVE switch test similar to Flexit_bacnet

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jbouwh committed:
Add validation to water_heater set_operation mode at entity component (#111168)

* Add validation to water_heater set_operation mode at entity component

* Add final decorator

starkillerOG committed:
Add Motionblinds virtual integration Acomax (#111458)

jbouwh committed:
Improve handling mqtt command template exceptions (#110499)

* Improve handling mqtt command template exceptions

* Fix test

* Cleanup stale exception handler

* Throw on topic template exception

starkillerOG committed:
Add Motionblinds virtual integration Krispol (#111456)

starkillerOG committed:
Add Motionblinds virtual integration Madeco (#111455)

emontnemery committed:
Improve logging of google_assistant messages (#110637)

* Improve logging of google_assistant messages

* Add tests

* Add test

Tronix117 committed:
Add @tronix117 as overkiz codeowner (#111464)

yozik04 committed:
Bump vallox-websocket-api to 5.1.0 (#111471)

Bump lib to 5.1.0

timmo001 committed:
Add title to reauthenticate integration issue (#111275)

* Add title to reauthenticate integration issue

* Fix translation placeholder in test

* Update translation placeholders in test_init.py

* Update integration reauthentication message

* Update string

Lash-L committed:
Bump Python-roborock to 0.39.2 (#111479)

bump to 0.39.2

autinerd committed:
Add webmin integration (#106976)

* add webmin integration 1

* refactor, add memory sensors

* Fix docstring

* addressed reviews

* address reviews

* address reviews

* use translation strings for sensors

* add async_abort_entries_match

* apply review comments

* address reviews

* add async_set_unique_id

* add identifiers to device_info

* disable all sensors by default

* move icons to icons.json

* show Faults when given from server in config flow

* add test for Fault

* Apply review suggestions

* Create helper functions for webmin instance and sorted mac addresses

* fix tests

bdraco committed:
Fix race in dlna_dmr test (#111428)

The entity was removed before the entity registry could update it

```
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity.py", line 1482, in _async_process_registry_update_or_remove
    assert registry_entry is not None
AssertionError

```

astrandb committed:
Bump myuplink lib to 0.5.0 (#111447)

abmantis committed:
Update idasen-ha to v2.5.1 to fix regression (#111439)

Update Idasen Desk support lib to fix regression

idasen v0.11.1 introduces some regressions, so idasen-ha v2.5.1 now
depends on v.0.11.0 instead.

bdraco committed:
Switch nmap_tracker to use aiooui (#111415)

mac-vendor-lookup can take a long time to startup because
it has to fetch the oui db from the web

bdraco committed:
Add support for eager tasks (#111425)

* Add support for eager tasks

python 3.12 supports eager tasks

reading:
https://docs.python.org/3/library/asyncio-task.html#eager-task-factory
https://github.com/python/cpython/issues/97696

There are lots of places were we are unlikely to suspend, but we might
suspend so creating a task makes sense

* reduce

* revert entity

* revert

* coverage

* coverage

* coverage

* coverage

* fix test

jpbede committed:
Remove `@bind_hass` from async_active_zone and make callback function (#111488)

bdraco committed:
Use discovery flow helper for hardware integrations (#111437)

* Use discovery flow helper for hardware integrations

The discovery flow helper defers loading discovered integrations until after startup
to improve startup reliability.

* Use discovery flow helper for hardware integrations

The discovery flow helper defers loading discovered integrations until after startup
to improve startup reliability. Since hardware was not listed in as a
discovery integration, the notification for new discoveries was missing.

farmio committed:
Update xknx to 2.12.1 and xknxproject to 3.7.0 (#111460)

jpbede committed:
Add `single_config_entry` manifest option (#109505)

* Allow setting if we support multiple config entries in config flow

* Move property to config flow instead of flow handler

* Move marking an integration as single instance only to manifest

* Revert line remove

* Avoid init a config flow or adding a new entry on a single instance with an entry

* Revert changes in test

* Process code review comments

* Apply code review suggestion

OnFreund committed:
Image entity media source (#104743)

* Image entity media source

* MJPEG streaming

* Update on change rather than fixed interval

* Only send boundary twice

* return when image has no data

* Write each frame twice

* Use friendly name when browsing

* Fix sending of double frame

* Initial image proxy test

* Improve proxy stream test

* Refactor

* Code review fixes

elupus committed:
Update rfxtrx library to handle connection retries (#111366)

Upgrade rfxtrx library to one that leaves reconnection to home assistant and handle loss of connection by reloading the integration.

https://github.com/Danielhiversen/pyRFXtrx/releases/tag/0.31.0

emontnemery committed:
Remove JuiceNet integration (#111477)

* Remove JuiceNet integration

* Adjust issue

* Add test

* Fix stale DOMAIN constant

lellky committed:
Add hvac_action to flexit_bacnet climate entity (#110498)

* Add hvac_action to flexit_bacnet climate entity

Also add extra state attributes

* Remove extra state attrs

* Change hvac action to heating

spycle committed:
Bump pyMicrobot to 0.0.17 (#111496)

* Update manifest.json

Up-version pyMicrobot to 0.0.17

* Update requirements_all.txt

Up-version pyMicrobot to 0.0.17

* Update requirements_test_all.txt

Up-version pyMicrobot to 0.0.17

starkillerOG committed:
Reolink trigger ReAuth when credentials invalid during update (#111492)

Check for CredentialsInvalidError

bdraco committed:
Add support for importing integrations in the executor (#111336)

* Add support for pre-imports at setup time

alternative solution to #111331

* refactor

* refactor

* refactor

* mark >1.0s integrations

* no point in executor if already loaded

* no point in executor if already loaded

* cleanup

* cleanup

* two more

* one more

* analytics loads a lot more integrations

* cloud

* debug

* psutil, hardwre

* try zha

* Update homeassistant/setup.py

* await

* comments

* coverage

* coverage

* coverage

* move logic to loader

* move logic to loader

* preserve comments

OnFreund committed:
Improve code comment for Chrome frame bug in Camera (#111504)

bdraco committed:
Fix dict changing size during iteration in deconz (#111523)

eclair4151 committed:
Add support to Lutron lights for color and temperature control (#109019)

* Add support to lutron lights for color and temperature control

* Fix review comments for adding color and temperature to Lutron lights

* Add eclair4151 as codeowner Lutron Caseta Integration

* Fix review comments for add color support for Lutron lights

* Code cleanup suggestions from PR review for adding color support for Lutron Lights

Co-authored-by: J. Nick Koston <nick@koston.org>

* Fix minor syntax errors from review comments for lutron light color support

* Fix review comments and code cleanup for Lutron light color support

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Import radio_browser in the executor to avoid blocking the event loop (#111519)

was not added to the original PR to contain scope

identified at: https://github.com/home-assistant/core/pull/111336#issuecomment-1963062529

bdraco committed:
Import influxdb in the executor to avoid blocking the event loop (#111520)

was not added to the original PR to contain scope

identified at: https://github.com/home-assistant/core/pull/111336#issuecomment-1963062529

jeeftor committed:
WeatherFlow Forecast (REST API) (#106615)

* rebase off dev

* Update homeassistant/components/weatherflow_cloud/const.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Addressing 1st round of PR Comments

* Update homeassistant/components/weatherflow_cloud/config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* addressing PR Comments

* fixing last comment that i can see

* Update homeassistant/components/weatherflow_cloud/coordinator.py

OOPS

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/weatherflow_cloud/weather.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/weatherflow_cloud/coordinator.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* switching to station id

* Update homeassistant/components/weatherflow_cloud/strings.json

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* addressing PR

* Updated tests to be better

* Updated tests accordingly

* REAuth flow and tests added

* Update homeassistant/components/weatherflow_cloud/strings.json

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/weatherflow_cloud/coordinator.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Addressing PR comments

* Apply suggestions from code review

* ruff fix

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: G Johansson <goran.johansson@shiftit.se>

bdraco committed:
Import mqtt in the executor to avoid blocking the event loop (#111521)

joostlek committed:
Add icon translations to Flo (#111542)

thecode committed:
Bump aioshelly to 8.0.2 (#111414)

jpbede committed:
Avoid call to `hass.helpers` in permobile (#111541)

niro1987 committed:
Add Tado add meter readings service (#111552)

joostlek committed:
Add icon translations to Fritz (#111547)

* Add icon translations to Fritz

* Add icon translations to Fritz

bdraco committed:
Fix text in docstrings and debug logging for loader (#111560)

synesthesiam committed:
Add speech-to-text cooldown for local wake word (#108806)

* Deconflict based on wake word

* Undo test

* Make wake up key a string, rename error

* Update snapshot

* Change to "wake word phrase" and normalize

* Move normalization into the wake provider

* Working on describe

* Use satellite info to resolve wake word phrase

* Add test for wake word phrase

* Match phrase with model name in wake word provider

* Check model id

* Use one constant wake word cooldown

* Update homeassistant/components/assist_pipeline/error.py

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

* Fix wake word tests

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Use an eager task to update multiple entities (#111556)

If there are multiple entities on the same platform its likely
that only one of them will suspend in the update so schedule
them eagerly

bdraco committed:
Create tasks eagerly in the discovery flow helper (#111557)

The likely outcome from most discoveries is an near immediate abort
so we run them eagerly to avoid having to schedule on the event loop
for the common case

bdraco committed:
Reduce latency to set up websocket forwarding in hassio (#111558)

Reduce latancy to set up websocket forwarding in hassio

Create the tasks eagerly to avoid one iteration of the event loop
to connect the proxy

bdraco committed:
Reduce latency to connect websocket writer (#111559)

Reduce latancy to connect websocket writer

Use an eager start task to start the writer

bdraco committed:
Convert person start to be a callback function (#111571)

Nothing was being awaited here so there was no need to create
a task per person at the start event.

The async_update_config coro remains since its required
by the collection but is now a wrapper around a callback
_async_update_config

bdraco committed:
Start the debouncer task eagerly (#111569)

bdraco committed:
Start automations with eager tasks (#111568)

bdraco committed:
Bump yalexs to 1.11.4 (#111567)

bdraco committed:
Make unifiprotect fallback update an eager task (#111580)

bdraco committed:
Use eager_start for sonos ping background task (#111578)

bdraco committed:
Refactor rainmachine to increase chance of reusing the connection (#111573)

bdraco committed:
Migrate gather_with_limited_concurrency to use create_eager_task (#111565)

bdraco committed:
Use eager start for august subscriber refreshes (#111581)

Since most of the time this call will be a noop since pubnub will be connected
we can avoid scheduling a task on the event loop every time it fires

bdraco committed:
Use eager task creation for entity platform polling (#111575)

* Use eager task creation for entity platform polling

We have lots of places where `async_update` does not suspend and
is only a coro because its required to be a coro to run in the
event loop

* try again

bdraco committed:
Import zeroconf in the executor to avoid blocking the event loop (#111591)

This one takes ~0.45s on the green

bdraco committed:
Import dhcp in the executor to avoid blocking the event loop (#111589)

Discovery integrations load a lot earlier when many modules
are not yet loaded so they have a lot less overlapping deps

dhcp uses scapy which is large to import. I hope one
day to replace it with something more lightweight

bdraco committed:
Import samsungtv in the executor to avoid blocking the event loop (#111588)

`2024-02-26 21:12:28.305 DEBUG (MainThread) [homeassistant.loader] Importing platform samsungtv.config_flow took 1.10s (loaded_executor=False)`

This one has a long requirements list

bdraco committed:
Import wemo in the executor to avoid blocking the event loop (#111587)

`2024-02-26 21:12:29.792 DEBUG (MainThread) [homeassistant.loader] Importing platform wemo.config_flow took 1.11s (loaded_executor=False)`

bdraco committed:
Import thread in the executor to avoid blocking the event loop (#111586)

Import thread in the executor

DEBUG (MainThread) [homeassistant.loader] Importing platform thread.config_flow took 2.92s (loaded_executor=False)

jesserockz committed:
ESPHome: Pass through wake_word_phrase from on-device microWakeWord (#111585)

* ESPHome: Pass through wake_word_phrase from on-device microWakeWord

* Bump aioesphomeapi to 23.0.0

bdraco committed:
Import spotify in the executor to avoid blocking the event loop (#111594)

bdraco committed:
Import august in the executor to avoid blocking the event loop (#111595)

bdraco committed:
Use an eager task in the update coordinator scheduled refresh (#111570)

* Use an eager task in the update coordinator scheduled refresh

We have a lot of places that will not suspend because the refresh function
decides it does not need to update. Currently these have to be scheduled
on the event loop even though they are a noop.

Since _handle_refresh_interval is subclassed in some integrations, I created
a dunder wrapper function to avoid integraions subclassing it

* fix time fires outside of patch

bdraco committed:
Create component setup tasks eagerly in bootstrap (#111511)

This saves waiting one iteration of the event loop

bdraco committed:
Import usb in the executor to avoid blocking the event loop (#111592)

`2024-02-26 21:38:00.091 DEBUG (MainThread) [homeassistant.loader] Component usb import took 0.753 seconds (loaded_executor=False)`

bdraco committed:
Fully fix race in removing entities (#111499)

Full fix race in removing entities

Followup to #110978 now that we can start the task
eagerly it completely closes the race window

bdraco committed:
Load config entry platforms eagerly (#111498)

We can avoid one event loop iteration to start the tasks here.

bdraco committed:
Create bootstrap tasks eagerly (#111497)

We can avoid one event loop iteration to start the tasks here

bdraco committed:
Remove zeroconf from ssdp after deps (#111326)

* Remove zeroconf from ssdp after deps

This was added in #36277 but is no longer needed since
we setup discovery integrations ahead of time to ensure
their deps are updated before other integrations can load
them

* adjust test

bdraco committed:
Reduce bootstrap code (#111450)

* Reduce bootstrap code

Adds a SETUP_ORDER const dict which will be used
for future refactoring

* tweak

* fix bad merge

bdraco committed:
Import ssdp in the executor to avoid blocking the event loop (#111593)

`2024-02-26 21:37:59.325 DEBUG (MainThread) [homeassistant.loader] Component ssdp import took 1.179 seconds (loaded_executor=False)`

ctalkington committed:
Update pyipp to 0.14.5 (#111590)

bdraco committed:
Refactor eafm to avoid creating entities in the coordinator update (#111601)

edenhaus committed:
Bump deebot-client to 6.0.2 (#111507)

bdraco committed:
Make core tests that should run in the event loop coros (#111605)

joostlek committed:
Add icon translations to Geocaching (#111613)

joostlek committed:
Add icon translations to Garages Amsterdam (#111610)

emontnemery committed:
Deprecate Logi Circle integration (#111491)

* Deprecate Logi Circle integration

* Update homeassistant/components/logi_circle/__init__.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

dependabot[bot] committed:
Bump actions/download-artifact from 4.1.2 to 4.1.3 (#111602)

cdce8p committed:
Update python-slugify to 8.0.4 (#111284)

cdce8p committed:
Update typing-extensions to 4.10.0 (#111538)

dmulcahey committed:
Add device list to ZHA config entry diagnostics (#111549)

* Add device list to ZHA  config entry diagnostics

* add logical type

bdraco committed:
Import zwave_js in the executor to avoid blocking the event loop (#111622)

`2024-02-27 10:55:54.872 DEBUG (MainThread) [homeassistant.loader] Component zwave_js import took 0.780 seconds (loaded_executor=False)`

brenank committed:
Add statistics keep_last_sample option (#88655)

* introduce preserve last value option

* improve comments

* add unit test

* skip scheduling purge on a preserved value

* do not schedule sensor update if preserving last value

* fix unit test to use new mock time pattern

pattern introduced in https://github.com/home-assistant/core/pull/93499

* rename preserve_last_val to keep_last_sample

* add keep_last_sample config validation

emontnemery committed:
Deprecate the Ambiclimate integration (#111627)

Deprecate Ambiclimate integration

edenhaus committed:
Deprecate mailbox platform (#110930)

TheNoim committed:
Extend ZHA thermostat local temperature calibration range for Sonoff TRVZB (#109691)

* Extend thermostat local temperature calibration range

* Extend thermostat local temperature calibration range only for TRVZB

* Change _attr_native_min_value and _attr_native_max_value back to -2.5 and 2.5

* Adjust _attr_native_step for TRVZB to 0.2

* Implement requested changes

* Fix platform for MULTI_MATCH

* Implement requested changes

* Implement requested changes

bdraco committed:
Import xbox in the executor to avoid blocking the event loop (#111625)

`Component xbox import took 0.326 seconds (loaded_executor=False)
`

bdraco committed:
Import airvisual in the executor to avoid blocking the event loop (#111624)

`Component airvisual import took 0.566 seconds (loaded_executor=False)`

bdraco committed:
Import shelly in the executor to avoid blocking the event loop (#111623)

`[homeassistant.loader] Component shelly import took 0.396 seconds (loaded_executor=False)`

joostlek committed:
Add icon translations to Fully kiosk (#111609)

* Add icon translations to Fully kiosk

* Update homeassistant/components/fully_kiosk/icons.json

bdraco committed:
Use an eager task to setup entity platforms (#111574)

* Use an eager task to setup entity platforms

Ideally we would have awaited this function instead, but we want
to shield it from cancellation so we wrap it in asyncio.shield
which schedules it as a task. Since we have integrations that
never suspend in async_setup_entry, we can avoid scheduling on
the evnet loop with an eager task for this case

* its an executor future

* its an executor future

* fix

* doc string lied

bdraco committed:
Ensure all data is set in seventeentrack before adding entities (#111603)

This integration would call async_add_entities before
it has set the data that the entity needed. There is
no guarantee that the entity would take long enough
to add that the data would be there

jpbede committed:
Avoid call to `hass.helpers.store` in LabelRegistry (#111555)

* Avoid call to `hass.helpers.store` in LabelRegistry

* Change type annotation

emontnemery committed:
Always allow ignore and unignore flows for single config entry integrations (#111631)

* Always allow ignore and unignore flows for single config entry integrations

* Update tests/test_config_entries.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

janrothkegel committed:
Wolflink connection pooling (#111483)

* Enable connection pooling

* Decrease polling frequency

* Replace client passing by instance instead of lambda

MartinHjelmare committed:
Fix rfxtrx flaky reconnect test (#111635)

synesthesiam committed:
Avoid race condition with UDP voice server starting in ESPHome (#111644)

* Avoid race condition with UDP server starting

* Fix test

ludeeus committed:
Move backup/* WS commands to the backup integration (#111636)

Revert "Revert "Move backup/* WS commands to the backup integration" (#111136)"

This reverts commit 8f834268955966d93bbc85b2ae7e86e6da46e37f.

bdraco committed:
Import bluetooth in the executor to avoid blocking the event loop (#111596)

emontnemery committed:
Update list of integrations known to violate color mode rules (#111647)

Kane610 committed:
Break out UniFi get_unifi_api function (#111323)

Break out get api function

emontnemery committed:
Return FlowResultType.ABORT when violating single_config_entry (#111637)

* Return FlowResultType.ABORT when violating single_config_entry

* Fix translations

* Fix tests

FedDam committed:
Add button platform to microBees (#111141)

* add button platform to microBees

* use list comprehension for async_add_entities

* add a transaltion_key and fix list comprehension

* add panic button

* remove BUTTON_PRODUCT_IDS

Kane610 committed:
Bump aiounifi to v72 (#111649)

bdraco committed:
Use eager task creation to add entities to entity platform (#111553)

catsmanac committed:
Bump pyenphase to 1.19.1 (#111653)

bdraco committed:
Bump aiohomekit to 3.1.5 (#111663)

thecode committed:
Fix Switcher test update CI failure (#111667)

mdegat01 committed:
Bad Request from supervisor should be error (#111162)

sairon committed:
Add ODROID-M1S as a supported OS hardware board (#111022)

joostlek committed:
Add icon translations to Goodwe (#111655)

joostlek committed:
Add icon translations to Google Sheets (#111661)

joostlek committed:
Add icon translations to Google Mail (#111660)

joostlek committed:
Add icon translations to Google Generative AI Conversation (#111659)

joostlek committed:
Add icon translations to Google Assistant (#111657)

catsmanac committed:
Add Grid import export to Enphase Envoy (#110884)

* Add Grid import export to enphase Envoy

* Update snapshot for labels dict element in entity registry

* use identity check for enum

* Revert use of identity check, didn't add entities

* Implement review feedback for tests

* ct phase sensors disabled by default

* import PHASENAMES from pyenphase

* Update tests/components/enphase_envoy/test_sensor.py

* Update tests/components/enphase_envoy/test_sensor.py

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

joostlek committed:
Add icon translations to Goalzero (#111630)

bdraco committed:
Use eager_start for shelly task creation (#111671)

jpbede committed:
Remove `@bind_hass` from `camera` functions (#111494)

* Remove `@bind_hass` from camera functions

* Pass hass to async_get_image in image_processing

Jc2k committed:
Cleanup BLE-only controls when migrating HomeKit BLE device to Thread (#110334)

Co-authored-by: J. Nick Koston <nick@koston.org>

jpbede committed:
Avoid call to `hass.helpers.store` in FloorRegistry (#111554)

* Avoid call to `hass.helpers.store` in FloorRegistry

* Change type annotation

bdraco committed:
Use eager_start for background service call tasks (#111620)

joostlek committed:
Add icon translations to Energyzero (#111513)

* Add icon translations to Energyzero

* Add icon translations to Energyzero

bdraco committed:
Migrate setup to use eager tasks (#111619)

bdraco committed:
Small speed up to Bluetooth setup (#111677)

bdraco committed:
Use eager_start for homekit task creation (#111675)

bdraco committed:
Await service calls in emulated_hue instead of making them blocking (#111674)

bdraco committed:
Update httpx to 0.27.0 and httpcore to 1.0.4 (#111681)

chemelli74 committed:
Bump aioshelly to 8.1.1 (#111678)

jpbede committed:
Drop `@bind_hass` use from hassio component (#111522)

* Drop `@bind_hass` use from hassio component

* Add comment why we import locally

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

joostlek committed:
Add icon translations to Google (#111656)

Co-authored-by: Dave T <17680170+davet2001@users.noreply.github.com>

joostlek committed:
Add icon translations to Foscam (#111546)

Co-authored-by: Dave T <17680170+davet2001@users.noreply.github.com>

Commit:
Bump pysignalclirestapi to 0.3.23 (#111501)

joostlek committed:
Add icon translations to Generic camera (#111612)

Co-authored-by: Dave T <17680170+davet2001@users.noreply.github.com>

bdraco committed:
Reduce task overhead in homekit_controller (#111670)

bdraco committed:
Reduce hassio setup time (#111686)

Reduce supervisor setup time

The two calls that take the most time are calling
push_config ~0.4s and updating the api ~0.35s and
can be run concurrently

bdraco committed:
Reduce latency in storage by making the tasks eager (#111500)

* Reduce latancy to load storage by making the task eager

This changes the semantics a bit under the hood because it
can raise sooner which means we do not store the task
as _load_task if it raises right away. That means
concurrent calls that result in failure are likely to try
again now which will be a tiny performance hit for this
case.

* fix

* will now finish in time

joostlek committed:
Add icon translations to Flume (#111543)

bdraco committed:
Avoid tracking tasks that finish right away (#111690)

bdraco committed:
Import xiaomi_ble in the executor to avoid blocking the event loop (#111693)

bdraco committed:
Make homekit_controller platform load tasks eagerly (#111701)

joostlek committed:
Add icon translations to Epson (#111528)

joostlek committed:
Add icon translations to Glances (#111615)

* Add icon translations to Glances

* Add icon translations to Glances

joostlek committed:
Add icon translations to GDACS (#111611)

* Add icon translations to GDACS

* Add icon translations to GDACS

* Add icon translations to GDACS

joostlek committed:
Add icon translations to FiveM (#111539)

joostlek committed:
Add icon translations to Environment Canada (#111527)

* Add icon translations to Environment Canada

* Add icon translations to Environment Canada

joostlek committed:
Add icon translations to Escea (#111529)

joostlek committed:
Add icon translations to Fireservicerota (#111537)

joostlek committed:
Add icon translations to Fastdotcom (#111534)

joostlek committed:
Add icon translations to Eufylife BLE (#111530)

joostlek committed:
Add icon translations to FAA delays (#111533)

joostlek committed:
Add icon translations to Google Assistant SDK (#111658)

bdraco committed:
Avoid creating system monitor disk sensors for non-dirs (#111695)

Avoid creating system monitor sensors for non-dirs

Currently we create sensors for /etc/hosts, /etc/asound.conf, since
they are bind mounts in the container. These all have to have
their own coordinator

joostlek committed:
Add icon translations to Filesize (#111536)

jpbede committed:
Revert "Remove `@bind_hass` from `camera` functions" (#111703)

Revert "Remove `@bind_hass` from `camera` functions (#111494)"

This reverts commit c890c1aeee7342e8fb290d434c054b27aa998b7e.

jpbede committed:
Revert "Remove `@bind_hass` from async_active_zone and make callback function" (#111704)

Revert "Remove `@bind_hass` from async_active_zone and make callback function…"

This reverts commit f85a8c504184b4787ec8d488814914ed42b0d794.

jpbede committed:
Revert "Drop `@bind_hass` use from hassio component (#111522)" (#111705)

This reverts commit 4e4345f04e7b4570bbd5dae7d68e20d104df929e.

joostlek committed:
Add icon translations to Denon avr (#111364)

* Add icon translations to Denon avr

* Update homeassistant/components/denonavr/icons.json

joostlek committed:
Add icon translations to Devolo home network (#111367)

erwindouna committed:
Tado code quality improvements (#107678)

Co-authored-by: Robert Resch <robert@resch.dev>

joostlek committed:
Add icon translations to Gree (#111662)

* Add icon translations to Gree

* Add icon translations to Gree

joostlek committed:
Add icon translations to Fronius (#111608)

swistakm committed:
Bump blebox_uniapi to 2.2.1 (#108918)

Co-authored-by: Robert Resch <robert@resch.dev>

bdraco committed:
Import mobile_app in the executor to avoid blocking the event loop (#111692)

2024-02-27 22:44:23.926 DEBUG (MainThread) [homeassistant.loader] Component mobile_app import took 0.490 seconds (loaded_executor=False)

bdraco committed:
Use eager tasks in a few more places in bootstrap (#111697)

I missed a few of the ones that run before we call
_async_set_up_integrations as I was too focused
on the timings of that function

bdraco committed:
Make entity component platform setup tasks eager (#111698)

Many of these that setup from YAML will never suspend
because they do no I/O

bdraco committed:
Use eager tasks when updating entity state after service calls (#111702)

In a multi-entity update case the state update may not need
to suspend and can get executed without scheduling a task
on the event loop

joostlek committed:
Add icon translations to DSMR (#111411)

* Add icon translations to DSMR

* Add icon translations to DSMR

joostlek committed:
Add icon translations to DROP connect (#111373)

* Add icon translations to DROP connect

* Add icon translations to DROP connect

joostlek committed:
Add icon translations to BMW (#111348)

* Add icon translations to BMW

* Add icon translations to BMW

joostlek committed:
Add icon translations to Guardian (#111712)

Co-authored-by: Aaron Bach <bachya1208@gmail.com>

chemelli74 committed:
Add Comelit humidifier/dehumidifier (#111203)

* Add Comelit humidifier/dehumidifier

* optimize turn_on

* fix entity naming

* raise ServiceValidationError

* apply review comment

* apply review comments

* rename HumidifierComelitAction

snoack committed:
Add entities for Balboa Spa pumps (#111245)

* Add entities for Balboa Spa pumps

* Fix fan tests and move client_update to __init__

* Ruff

---------

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

joostlek committed:
Add icon translations to HVV departures (#111728)

Kane610 committed:
Adapt Axis integration to library refactoring (#110898)

* Adapt Axis integration to library refactoring

* Bump axis to v49

dependabot[bot] committed:
Bump github/codeql-action from 3.24.4 to 3.24.5 (#111448)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.24.4 to 3.24.5.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.24.4...v3.24.5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bdraco committed:
Import tplink in the executor to avoid blocking the event loop (#111691)

* Import tplink in the executor to avoid blocking the event loop

2024-02-27 22:44:19.908 DEBUG (MainThread) [homeassistant.loader] Component tplink import took 1.620 seconds (loaded_executor=False)

* patch out discovery because it happens too fast now

bdraco committed:
Import nexia in the executor to avoid blocking the event loop (#111694)

2024-02-27 22:54:18.638 DEBUG (MainThread) [homeassistant.loader] Importing platform nexia.config_flow took 1.84s (loaded_executor=False)

emontnemery committed:
Allow translating device names (#110711)

* Allow translating device names

* Don't keep a reference to translations in config entry

* Update kitchen_sink tests

* Add tests

snoack committed:
Add entities for Balboa Spa lights (#111252)

joostlek committed:
Add icon translations to Homematic IP Cloud (#111723)

joostlek committed:
Add icon translations to Homekit Controller (#111720)

* Add icon translations to Homekit Controller

* Add icon translations to Homekit Controller

joostlek committed:
Add icon translations to Home connect (#111718)

joostlek committed:
Add icon translations to Hive (#111717)

joostlek committed:
Add icon translations to Heos (#111716)

joostlek committed:
Add icon translations to Hassio (#111715)

joostlek committed:
Add icon translations to Harmony (#111714)

joostlek committed:
Add icon translations to Homekit (#111719)

Augar committed:
Add ecovacs lawn mover (#111673)

joostlek committed:
Add icon translations to Huawei LTE (#111724)

* Add icon translations to Huawei LTE

* Add icon translations to Huawei LTE

joostlek committed:
Add icon translations to Huisbaasje (#111726)

* Add icon translations to Huisbaasje

* Add icon translations to Huisbaasje

bdraco committed:
Remove gather for loading generated files in zeroconf (#111699)

async_get_custom_components is the only place this can suspend
and it will always be cached by the time this is called so there
is no reason to schedule these on the event loop

bdraco committed:
Use eager tasks in ssdp (#111696)

There are a few places where we are likely to never suspend here

bramkragten committed:
Update frontend to 20240228.0 (#111736)

miaucl committed:
Cleanup swiss_public_transport error logging (#111741)

bdraco committed:
Import plex in the executor to avoid blocking the event loop (#111743)

`Importing platform plex.config_flow took 0.54s (loaded_executor=False)`

bdraco committed:
Eagerly unload config entries in tests (#111707)

This is a small latancy reduction to reduce test wall clock time
since it runs every test

bdraco committed:
Import powerwall in the executor to avoid blocking the event loop (#111749)

`2024-02-27 22:44:17.087 DEBUG (MainThread) [homeassistant.loader] Component powerwall import took 0.459 seconds (loaded_executor=False)`

cottsay committed:
Bump aioraven to 0.5.1 (#111600)

puddly committed:
Use new zigpy OTA providers for ZHA (#111159)

* Use `None` instead of `"unknown"` when the current version is unknown

* Only use the current file version from the OTA notification

* Use `sw_version`, if available, and update `current_file_version`

* Assume the current version is the latest version

* Fix lint errors

* Use `image` instead of `firmware`

* Include a changelog if updates expose it

* Clear latest firmware only after updating the installed version

* Bump minimum zigpy version to 0.63.0

* Create a data update coordinator to consolidate updates

* Fix overridden `async_update`

* Fix most unit tests

* Simplify `test_devices` to fix current tests

* Use a dict comprehension for creating mocked entities

* Fix unit tests (thanks @dmulcahey!)

* Update the currently installed version on cluster attribute update

* Drop `PARALLEL_UPDATES` now that we use an update coordinator

* Drop `_reset_progress`, it is already handled by the update component

* Do not update the progress if we are not supposed to be updating

* Ignore latest version (e.g. if device attrs changed) if zigpy rejects it

* Clean up handling of command id in `Ota.cluster_command`

* Start progress at 1%: 0 and False are considered equal and are filtered!

Use `ceil` instead of remapping 1-100

* The installed version will be auto-updated when the upgrade succeeds

* Avoid 1 as well, it collides with `True`

* Bump zigpy to (unreleased) 0.63.2

* Fix unit tests

* Fix existing unit tests

Send both event types

Globally enable sending both event types

* Remove unnecessary branches

* Test ignoring invalid progress callbacks

* Test updating a device with a no longer compatible firmware

Augar committed:
Add Ecovacs goat switch entities (#111751)

yanuino committed:
Read min/max number of showers from state for DomesticHotWaterProduction in Overkiz integration (#111535)

* Read min/max number of showers from state

* Rewrite code for Read min/max number of showers from state

* Set _attr_ instead of inherited value

IsakNyberg committed:
Add permobil entity (#111761)

* add permobil entity

* small fixes

* remove sensor init

Tronix117 committed:
Add  overkiz support for Atlantic Shogun ZoneControl 2.0 (AtlanticPassAPCHeatingAndCoolingZone) (#110510)

* Add Overkiz support for AtlanticPassAPCHeatingAndCoolingZone widget

* Add support for AUTO HVAC mode for Atlantic Pass APC ZC devices that support it

* Add support for multiple IO controllers for same widget (mainly for Atlantic APC)

* Implement PR feedback

* Small PR fixes

* Fix constant inversion typo

puddly committed:
Improve ZHA group color modes (#111669)

* Set the color mode based on supported color modes

* Replace `zha` with `tuya` in unit test

bdraco committed:
Import isy994 in the executor to avoid blocking the event loop (#111766)

bramkragten committed:
Bump version to 2024.4.0.dev0 (#111755)

bdraco committed:
Import opower in the executor to avoid blocking the event loop (#111778)

fixes #111777

bdraco committed:
Import coinbase in the executor to avoid blocking the event loop (#111774)

fixes #111773

bdraco committed:
Import blink in the executor to avoid blocking the event loop (#111772)

fixes #111771

marcelveldt committed:
Bump aiohue to 4.7.1 (#111770)

bump aiohue to 4.7.1

bdraco committed:
Import cryptography early since importing openssl is not thread-safe (#111768)

cdce8p committed:
Improve zha coordinator typing (#111767)

marcelveldt committed:
Bump Python Matter Server to 5.7.0 (#111765)

bdraco committed:
Import androidtv_remote in the executor to avoid blocking the event loop (#111776)

fixes #111775

bdraco committed:
Move DATA_LOGGING constant to homeassistant.const (#111763)

bdraco committed:
Import backup in the executor to avoid blocking the event loop (#111781)

jpbede committed:
Add normalized name registry items base class (#111666)

* Add normalized name base registry items class

* Add tests

bdraco committed:
Bump securetar to 2024.2.1 (#111782)

bdraco committed:
Fix time trigger tests with leap year (#111785)

bdraco committed:
Import analytics_insights in the executor to avoid blocking the event loop (#111786)

fixes #111780

bdraco committed:
Pre-import api, config, and lovelace in bootstrap to avoid loading them at runtime (#111752)

balloob committed:
Remove 'values' from services validation (#111440)

* Remove 'values' from services validation

* Remove extra blank line

bdraco committed:
Fix steamist blocking startup by waiting for discovery (#111789)

Fix steamist blocking statup by waiting for discovery

bdraco committed:
Import discord in the executor to avoid blocking the event loop (#111790)

`2024-02-28 19:20:04.485 DEBUG (MainThread) [homeassistant.loader] Component discord import took 1.181 seconds (loaded_executor=False)`

bdraco committed:
Fix flux_led blocking startup by waiting for discovery (#111787)

* Avoid blocking startup by waiting for discovery in flux_led

* remove started discovery

bdraco committed:
Fix tplink blocking startup by waiting for discovery (#111788)

* Fix tplink blocking statup by waiting for discovery

* remove started

allenporter committed:
Fix calendar trigger to survive config entry reloads (#111334)

* Fix calendar trigger to survive config entry reloads

* Apply suggestions from code review

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
get_matter_device_info: Test the Matter config entry is set up (#111792)

Ensure the Matter config entry is set up

Noltari committed:
Use proper constant in Airzone Cloud climate tests (#111747)

tests: airzone_cloud: avoid using airzone const

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Kane610 committed:
Axis improve coverage binary tests (#111758)

* Parametrize binary sensor tests

* Add test coverage to the different *guard apps

* Add object analytics tests

synesthesiam committed:
Bump intents and add sentence tests (#111791)

joostlek committed:
Add icon translations to Hue (#111725)

bdraco committed:
Import enphase_envoy in the executor to avoid blocking the event loop (#111805)

jpbede committed:
Deprecate `hass.components` and log warning if used inside custom component (#111508)

* Deprecate @bind_hass and log error if used inside custom component

* Log also when accessing `hass.components`

* Log warning only when `hass.components` is used

* Change version

* Process code review

emontnemery commented:
@t3hk0d3 You have made some mistake when merging or rebasing your work.

I find the workflow in this guide very clear and easy to use to avoid such mistakes: <https://gist.github.com/Chaser324/ce0505fbed06b947d962>

Please open a new PR, targeting our dev branch, with only the relevant changes 👍
